src/System.Memory/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -248,52 +292,174 @@ public static OperationStatus DecodeFromUtf8InPlace(Span<byte> buffer, out int b
 
                 if (i0 < 0)
                     goto InvalidExit;
-                WriteThreeLowOrderBytes(ref Unsafe.Add(ref bufferBytes, destIndex), i0);
+
+                WriteThreeLowOrderBytes(ref Unsafe.Add(ref bufferBytes, (IntPtr)destIndex), i0);
                 destIndex += 3;
             }
-            else if (i2 != EncodingPad)
+            else if (t2 != EncodingPad)
             {
-                i2 = Unsafe.Add(ref decodingMap, i2);
+                int i2 = Unsafe.Add(ref decodingMap, (IntPtr)t2);
 
                 i2 <<= 6;
 
                 i0 |= i2;
 
                 if (i0 < 0)
                     goto InvalidExit;
-                Unsafe.Add(ref bufferBytes, destIndex) = (byte)(i0 >> 16);
-                Unsafe.Add(ref bufferBytes, destIndex + 1) = (byte)(i0 >> 8);
+
+                Unsafe.Add(ref bufferBytes, (IntPtr)destIndex) = (byte)(i0 >> 16);
+                Unsafe.Add(ref bufferBytes, (IntPtr)(destIndex + 1)) = (byte)(i0 >> 8);
                 destIndex += 2;
             }
             else
             {
                 if (i0 < 0)
                     goto InvalidExit;
-                Unsafe.Add(ref bufferBytes, destIndex) = (byte)(i0 >> 16);
+
+                Unsafe.Add(ref bufferBytes, (IntPtr)destIndex) = (byte)(i0 >> 16);
                 destIndex += 1;
             }
 
         DoneExit:
-            bytesWritten = destIndex;
+            bytesWritten = (int)destIndex;
             return OperationStatus.Done;
 
         InvalidExit:
-            bytesWritten = destIndex;
+            bytesWritten = (int)destIndex;
             return OperationStatus.InvalidData;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void Avx2Decode(ref byte src, ref byte destBytes, int sourceLength, int destLength, ref uint sourceIndex, ref uint destIndex)
+        {
+            ref byte srcStart = ref src;
+            ref byte destStart = ref destBytes;
+            ref byte simdSrcEnd = ref Unsafe.Add(ref src, (IntPtr)((uint)sourceLength - 45 + 1));
+
+            // The JIT won't hoist these "constants", so help him
```

Comment:
nit: `so help him.` -> `so help it.`

Reply:
Done -- will be updated with the next push.